static/app/views/performance/newTraceDetails/guards.tsx

```diff
@@ -0,0 +1,52 @@
+import {
+  ParentAutogroupNode,
+  SiblingAutogroupNode,
+  type TraceTree,
+  type TraceTreeNode,
+} from './traceTree';
+
+export function isMissingInstrumentationNode(
+  node: TraceTreeNode<TraceTree.NodeValue>
+): node is TraceTreeNode<TraceTree.MissingInstrumentationSpan> {
+  return !!(
+    node.value &&
+    'type' in node.value &&
+    node.value.type === 'missing_instrumentation'
+  );
+}
+
+export function isSpanNode(
+  node: TraceTreeNode<TraceTree.NodeValue>
+): node is TraceTreeNode<TraceTree.Span> {
+  return !!(node.value && ('description' in node.value || 'span_id' in node.value));
```

Comment:
Transactions have span_ids too, but no description. we should just stick to `'description' in node.value`

Reply:
Spans cant have description field right?

Reply:
transactions can't, all spans do.